src/vs/workbench/contrib/positronOutputWebview/browser/notebookOutputWebviewServiceImpl.ts

```diff
@@ -420,8 +453,48 @@ window.onload = function() {
 	 */
 	static readonly CssAddons = `
 <style>
-	/* Hide actions button that does things like opening source code etc.. (See #2829) */
-	.vega-embed details[title="Click to view actions"] {display: none;}
+	/* Hide actions button that try and open external pages like opening source code as they don't currently work (See #2829)
+	/* We do support download link clicks, so keep those. */
+	.vega-actions a:not([download]) {
+		display: none;
+	}
 </style>`;
 
+
+	private async _downloadData(
+		payload: PositronDownloadMessage,
+	): Promise<void> {
+		if (typeof payload.data !== 'string') {
+			return;
+		}
+
+		const [splitStart, splitData] = payload.data.split(';base64,');
+		if (!splitData || !splitStart) {
+			return;
+		}
+
+		const defaultDir = this._workspaceContextService.getWorkspace().folders[0]?.uri ?? await this._fileDialogService.defaultFilePath();
+		let defaultName: string;
+		if (payload.downloadName) {
+			defaultName = payload.downloadName;
+		} else {
+			const mimeType = splitStart.replace(/^data:/, '');
+			const candidateExtension = mimeType && getExtensionForMimeType(mimeType);
+			defaultName = candidateExtension ? `download${candidateExtension}` : 'download';
+		}
+
+		const defaultUri = joinPath(defaultDir, defaultName);
+		const newFileUri = await this._fileDialogService.showSaveDialog({
+			defaultUri
+		});
+		if (!newFileUri) {
+			return;
+		}
+
+		const buff = decodeBase64(splitData);
+		await this._fileService.writeFile(newFileUri, buff);
+		await this._openerService.open(newFileUri);
```

Comment:
Currently, the save dialog doesn't open the plot after its saved. Should this have the same behavior?